### PR TITLE
feat: supports selecting file version

### PR DESCRIPTION
**Pull request by 0xff-dev**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it
feat: supports selecting file version

#### Which issue(s) this PR fixes

Fix https://github.com/kubeagi/arcadia/issues/932

### CODE REVIEW

File: apiserver/pkg/versioneddataset/versioned_dataset.go [hunk 1]
```diff
@@ -300,14 +311,26 @@ func CreateVersionedDataset(ctx context.Context, c client.Client, input *generat
...
+				tmpFv := v1alpha1.FileWithVersion{
+					Path:    fv.Path,
+					Version: "",
+				}
+				if fv.Version != nil {
+					tmpFv.Version = *fv.Version
+				}
+				tmp.Paths = append(tmp.Paths, tmpFv)
+			}
+			// }
```
Comment (Abirdcfly) [inline apiserver/pkg/versioneddataset/versioned_dataset.go:332]: nit: remove this line

File: controllers/base/knowledgebase_controller.go [hunk 2]
```diff
@@ -451,33 +451,34 @@ func (r *KnowledgeBaseReconciler) reconcileFileGroup(ctx context.Context, log lo
...
 				Phase:          arcadiav1alpha1.FileProcessPhasePending,
 				ErrMessage:     "",
+				Version:        path.Version,
 			})
 			fileDetail = &fileGroupDetail.FileDetails[len(fileGroupDetail.FileDetails)-1]
 		}
 
 		switch strings.ToLower(group.Source.Kind) {
 		case "versioneddataset":
-			// info.Object has been
```
Comment (Abirdcfly) [inline controllers/base/knowledgebase_controller.go:474]: complate it?

**Review by Abirdcfly — APPROVED** [review]

File: api/base/v1alpha1/datasource_types.go [hunk 3]
```diff
@@ -55,6 +55,8 @@ type OSS struct {
 	Bucket string `json:"bucket,omitempty"`
 	// Object must end with a slash "/" if it is a directory
 	Object string `json:"object,omitempty"`
+
+	VersionID string `json:"versionID,omitempty"`
```
Comment (bjwswang) [inline api/base/v1alpha1/datasource_types.go:59]: Add a comment here to explain what is `VersionID` for?

File: apiserver/pkg/knowledgebase/knowledgebase.go [hunk 4]
```diff
@@ -269,9 +276,11 @@ func CreateKnowledgeBase(ctx context.Context, c client.Client, input generated.C
 			}
 			fileDetails := make([]*generated.Filedetail, len(fg.Paths))
 			for findex, path := range fg.Paths {
+				fmt.Println(path)
```
Comment (bjwswang) [inline apiserver/pkg/knowledgebase/knowledgebase.go:279]: fmt -> klog

File: controllers/base/knowledgebase_controller.go [hunk 5]
```diff
@@ -139,6 +139,10 @@ func (r *KnowledgeBaseReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, nil
 	}
 
+	if migrated := r.migratePaths2Files(ctx, kb); migrated {
```
Comment (bjwswang) [inline controllers/base/knowledgebase_controller.go:142]: add comments to indicate why this migration is required

File: controllers/base/knowledgebase_controller.go [hunk 6]
```diff
@@ -669,3 +673,30 @@ func (r *KnowledgeBaseReconciler) isReady(kb *arcadiav1alpha1.KnowledgeBase) boo
 	v, ok := r.ReadyMap[string(kb.GetUID())]
 	return ok && v
 }
+
+func (r *KnowledgeBaseReconciler) migratePaths2Files(ctx context.Context, kb *arcadiav1alpha1.KnowledgeBase) bool {
```
Comment (bjwswang) [inline controllers/base/knowledgebase_controller.go:677]: add comments on this func

File: apiserver/pkg/knowledgebase/knowledgebase.go [hunk 7]
```diff
@@ -75,11 +75,11 @@ func knowledgebase2model(ctx context.Context, c client.Client, knowledgebase *v1
 		if fg.Source.Namespace != nil {
 			ns = *fg.Source.Namespace
 		}
-		for in, fn := range fg.Paths {
-			key := fmt.Sprintf("%s/%s/%s", ns, fg.Source.Name, fn)
+		for in, fn := range fg.Files {
```
Comment (bjwswang) [inline apiserver/pkg/knowledgebase/knowledgebase.go:78]: well... if user still havs some files in `fg.Paths`, we should also return it.

**Review by bjwswang — APPROVED** [review]

**Pull request merged**